src/bin/ambit/cmd.rs

```diff
@@ -0,0 +1,191 @@
+// Symlink function is dependent on OS
+#[cfg(unix)]
+use std::os::unix::fs::symlink;
+#[cfg(windows)]
+use std::os::windows::fs::symlink_file as symlink;
+use std::{
+    fs,
+    io::{self, Write},
+    process::Command,
+};
+
+use ambit::{
+    config,
+    error::{AmbitError, AmbitResult},
+};
+
+use crate::directories::{AmbitPath, AmbitPathKind, AMBIT_PATHS};
+
+// Initialize config and repository directory
+fn ensure_paths_exist(force: bool) -> AmbitResult<()> {
+    if !AMBIT_PATHS.config.exists() {
+        AMBIT_PATHS.config.create()?;
+    }
+    if AMBIT_PATHS.repo.exists() && !force {
+        // Dotfile repository should not be overwritten if force is false
+        return Err(AmbitError::Other(
+            "Dotfile repository already exists.\nUse '-f' flag to overwrite.".to_owned(),
+        ));
+    } else if AMBIT_PATHS.repo.exists() {
+        // Repository directory exists but force is enabled
+        AMBIT_PATHS.repo.remove()?;
+    }
+    Ok(())
+}
+
+// Fetch entries from config file and return as vector
+fn get_config_entries() -> AmbitResult<Vec<config::Entry>> {
+    let content = AMBIT_PATHS.config.as_string()?;
+    config::get_entries(content.chars().peekable())
+        .collect::<Result<Vec<_>, _>>()
+        .map_err(AmbitError::Parse)
+}
+
+// Initialize an empty dotfile repository
+pub fn init(force: bool) -> AmbitResult<()> {
+    ensure_paths_exist(force)?;
+    AMBIT_PATHS.repo.create()?;
+    // Initialize an empty git repository
+    git(vec!["init"])?;
+    Ok(())
+}
+
+// Clone an existing dotfile repository with given origin
+pub fn clone(force: bool, arguments: Vec<&str>) -> AmbitResult<()> {
+    ensure_paths_exist(force)?;
+    // Clone will handle creating the repository directory
+    let repo_path = AMBIT_PATHS.repo.to_str()?;
+    let status = Command::new("git")
+        .arg("clone")
+        .args(arguments)
+        // Pass in ambit repo path as last argument to ensure that it is always cloned to the known path
+        .arg(repo_path)
```

Comment:
The `repo_path` argument should be prefixed with `--` so `git` does not think it is an option. (Not doing so can be problematic if e.g. the user sets the repo to ~/`-config`.)